File: SampleProjects/AutoLayout/AutoLayoutSample/AutoLayoutSample/ViewController.m
//
//  ViewController.m
//  AutoLayoutSample
//
//  Created by yuichi.takeda on 3/5/15.
//  Copyright (c) 2015 mixi, Inc. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (strong, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)setShortText:(id)sender {
    self.label.text = @"Hello, World!!";
}

- (IBAction)setNormalText:(id)sender {
    self.label.text = @"秋の田のかりほの庵の苫をあらみ　わが衣手は露にぬれつつ";
}

- (IBAction)setLongText:(id)sender {
    self.label.text = @"あのイーハトーヴォのすきとおった風、夏でも底に冷たさをもつ青いそら、うつくしい森で飾られたモリーオ市、郊外のぎらぎらひかる草の波。\nまたそのなかでいっしょになったたくさんのひとたち、ファゼーロとロザーロ、羊飼のミーロや、顔の赤いこどもたち、地主のテーモ、山猫博士のボーガント・デストゥパーゴなど、いまこの暗い巨きな石の建物のなかで考えていると、みんなむかし風のなつかしい青い幻燈のように思われます。では、わたくしはいつかの小さなみだしをつけながら、しずかにあの年のイーハトーヴォの五月から十月までを書きつけましょう。";
}

@end

File: Swift/samples/day3/sample3-3/KVOSample/KVOSample/ViewController.swift
//
//  ViewController.swift
//  KVOSample
//
//  Created by 鈴木大貴 on 2016/10/30.
//  Copyright © 2016年 marty-suzuki. All rights reserved.
//

import UIKit

class Counter: NSObject {
    static let shared = Counter()
    
    @objc dynamic var count: Int = 0
    
    private override init() {
        super.init()
    }
}

class ViewController: UIViewController {

    @IBOutlet weak var countLabel: UILabel!
    
    deinit {
        Counter.shared.removeObserver(self, forKeyPath: #keyPath(Counter.count))
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        countLabel.text = "\(Counter.shared.count)"
        
        Counter.shared.addObserver(self, forKeyPath: #keyPath(Counter.count), options: [.new, .old], context: nil)
    }

    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
        switch (keyPath, change?[.newKey] as? Int) {
        case ((#keyPath(Counter.count))?, let newValue?):
            countLabel.text = "\(newValue)"
            
        default:
            break
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func countDownButtonTapped(_ sender: UIButton) {
        Counter.shared.count = max(0, Counter.shared.count - 1)
    }
    
    @IBAction func countUpButtonTapped(_ sender: UIButton) {
        Counter.shared.count += 1
    }

    @IBAction func pushButtonTapped(_ sender: UIButton) {
        guard let viewController = storyboard?.instantiateViewController(withIdentifier: "ViewController") else { return }
        navigationController?.pushViewController(viewController, animated: true)
    }
}

